objects/src/transaction/tx_result.rs

```diff
@@ -266,3 +273,75 @@ fn extract_account_storage_delta(
 
     Ok(storage_delta)
 }
+
+// ACCOUNT VAULT DELTA
+// ================================================================================================
+// TODO: update when TMST depth 64 leaves are supported
+/// Extracts the account vault delta between the `initial_account` and `final_account_stub` from
+/// the provided `MerkleStore` and `BTreeMap`.
+fn extract_account_vault_delta(
+    store: &MerkleStore,
+    value_map: &BTreeMap<[u8; 32], Vec<Felt>>,
+    initial_account: &Account,
+    final_account_stub: &FinalAccountStub,
+) -> Result<AccountVaultDelta, TransactionResultError> {
+    // extract original assets from the vault
+    let mut orig_assets = initial_account
+        .vault()
+        .assets()
+        .map(|asset| (Digest::from(asset.vault_key()), asset))
+        .collect::<BTreeMap<_, _>>();
+
+    // extract final assets in the vault from the merkle store and advice map
+    let final_leaves = store
+        .non_empty_leaves(final_account_stub.0.vault_root(), 64)
+        .map(|(_, leaf)| leaf)
+        .collect::<Vec<_>>();
+    let final_assets = final_leaves
+        .into_iter()
+        .map(|leaf| {
+            let data =
+                value_map.get(&leaf.as_bytes()).expect("asset node must exist in the value map");
+            let asset = Asset::try_from(Word::try_from(&data[4..]).expect("data contains word"))
+                .expect("asset is well formed");
+            (Digest::from(asset.vault_key()), asset)
```

Comment:
We use `Digest::from(asset.vault_key())` because the `BTreeMap` needs the key to implement `Ord`, this is true of `Digest` however `Word` (the type of `vault_key()`) does not implement `Ord`.